analyzer/config/yaml/yaml_test.go

```diff
@@ -0,0 +1,108 @@
+package yaml
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/open-policy-agent/conftest/parser/yaml"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/analyzer/config"
+	"github.com/aquasecurity/fanal/types"
+)
+
+func Test_yamlConfigAnalyzer_Analyze(t *testing.T) {
+	tests := []struct {
+		name      string
+		inputFile string
+		want      *analyzer.AnalysisResult
+		wantErr   string
+	}{
+		{
+			name:      "happy path",
+			inputFile: "testdata/deployment.yaml",
+			want: &analyzer.AnalysisResult{
+				Configs: []types.Config{
+					{
+						Type:     config.YAML,
+						FilePath: "testdata/deployment.yaml",
+						Content: map[string]interface{}{
+							"apiVersion": "apps/v1",
+							"kind":       "Deployment",
+							"metadata": map[string]interface{}{
+								"name": "hello-kubernetes",
+							},
+							"spec": map[string]interface{}{
+								"replicas": float64(3),
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:      "broken YAML",
+			inputFile: "testdata/broken.yaml",
+			wantErr:   "unmarshal yaml",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b, err := ioutil.ReadFile(tt.inputFile)
+			require.NoError(t, err)
+
+			a := yamlConfigAnalyzer{
+				parser: &yaml.Parser{},
+			}
+
+			got, err := a.Analyze(analyzer.AnalysisTarget{
+				FilePath: tt.inputFile,
+				Content:  b,
+			})
+
+			if tt.wantErr != "" {
+				require.NotNil(t, err)
+				assert.Contains(t, err.Error(), tt.wantErr)
+				return
+			}
+			assert.NoError(t, err)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func Test_yamlConfigAnalyzer_Required(t *testing.T) {
```

Comment:
If there is a possibility of changing the yaml library in the future, I think it is better to test the required implementation.

e.g. 
```
hoge: &hoge
  line: single line
fuga:
  <<: *hoge
  comment: |
    multi
    line
```